src/main/scala/io/iohk/ethereum/mpt/Node.scala

```diff
@@ -62,10 +72,50 @@ case class BranchNode(children: Seq[Option[Either[ByteString, MptNode]]], termin
     */
   def updateChild(childIndex: Int, childNode: MptNode): BranchNode = {
     val childCapped = childNode.capped
-    BranchNode(children.updated(childIndex, Some(if (childCapped.length == 32) Left(childCapped) else Right(childNode))), terminator)
+    val updatedChild = if (childCapped.length == 32) HashNode(childCapped) else childNode
```

Comment:
also `if (childCapped.length == 32) HashNode(childCapped) else childNode` structure is used 3 times (on 129 and 160 line)